include/votca/csg/topology.h

```diff
@@ -447,9 +447,8 @@ inline Bead *Topology::CreateBead(Bead::Symmetry symmetry, std::string name,
 }
 
 inline Molecule *Topology::CreateMolecule(std::string name) {
-  Molecule *mol = new Molecule(_molecules.size(), name);
-  _molecules.push_back(mol);
-  return mol;
+  _molecules.emplace_back(new Molecule(_molecules.size(), name));
```

Comment:
I think here just using `(_molecules.size(), name)` could be enough, because emplace_back forwards the things to the constructor. Otherwise use `make_unique` normally no `new` should appear in the code

Reply:
I cannot get either of your suggestions to compile. The first one is because _molecules now stores unique pointers so it is expecting the argument to create a unique pointer. The second one becomes really long because I will also have to call move.

Reply:
```suggestion
  _molecules.emplace_back(mstd::make_unique<Molecule>(_molecules.size(), name));
```